### DISCUSSION TITLE
Simple question about SRP

### DISCUSSION THREAD

**Discussion by Exormeter** [openthread/openthread#9374] [Q&A] [answered]
Hello, I am new to openthread and I have a question about SRP and if it the right solution to the problem I am facing.
The current setup is a single Raspbeery Pi as the border router and a nrf52840 as a node. The border router is also currently a coap server and the nrf is a coap client, for the client I have gotten the zephyr coap client example working. 

Currently, the ip address of the server is hard coded into the example, which of course is not practical, so I want to use a host name instead.

Would setting up the Raspberry Pi as a SRP server and the nrf as a srp client allow me to have a look up of the ip address via a host name? Or does the Rpi need to be a client and the node in the network need to be a server? I am abit confused about how is discovering who.

**Comment by abtink** [comment, marked as answer]
There are different components that work together to make service discovery possible:

- SRP (Service Registration Protocol): SRP clients register the services they provide along with their hostname (and their IPv6 addresses) with the SRP server.
- DNS client: A DNS client can send query messages to browse or discover/resolve services and/or resolve host addresses given a hostname.
- DNS-SD  server/resolver: A DNS-SD server/resolver responds to such queries.

BRs (Border Routers) are typically configured to act as SRP servers and DNS-SD servers. Any device can be a DNS client and an SRP client.

Here is what I would suggest for your use case:

- The node acting as the CoAP server can register an SRP service indicating that it provides the "coap service" (which includes a hostname and registers its addresses).
- Other devices can then act as DNS clients to query and discover this service, resolve the host address providing this service, and then connect to it.
